Makefile

```diff
@@ -13,6 +13,40 @@ $(warning Undefining GOFLAGS set in CI)
 undefine GOFLAGS
 endif
 
+# Tools #
+
+# The tools required to build and test the project come from two sources:
+# 1. .bingo/Variables.mk: tools that are orthogonal to OLM, e.g.
+#   - golangci-lint
+#   - helm
+#   - kind
+#   - setup-envtest
+#   - yq
+# 2. tools.go: imports modules that need to be vendored and/or have a low tolerance for skew with the main module
+#              or because they are already imported by the main module, but we want to pull down the whole dependency
+#              to run some cmd or script, or get some resource (e.g. CRD yaml files).
+#   - OLM API CRDs
+#   - Test libraries, e.g. ginkgo
+#   - Code generation tools, e.g. k8s.io/code-generator
+
+# bingo manages the type 1 tools. If
+#  a) we don't want their dependencies affecting ours, and
+#  b) the tool's version doesn't need to track closely with OLM
+# the tool goes here
+include .bingo/Variables.mk
+
+# tools.go manages type 2 tools. If
+# a) we need some resource (e.g. yaml file)
+# b) need to run a script
+# c) need to track tools closely with OLM
+# the tools belongs in tools.go
+TOOL_EXEC := GO111MODULE=on go run -mod=vendor
+GINKGO := $(TOOL_EXEC) github.com/onsi/ginkgo/v2/ginkgo
+
+.PHONY: g
+g:
+	$(GINKGO)
```

Comment:
I'm not following - why do we need this one?

Reply:
oh whoops - that was a test target!!